SU-HSDO/suhumsci
# STN-117: Masthead

# READY FOR REVIEW

## Summary
This pull request adds the basic layout and styles for the masthead section.  For now we are going with the default [Decanter styles](https://decanter.stanford.edu/component/identity-masthead/) until design is approved.

**Desktop**
<img width="1268" alt="Screen Shot 2020-02-17 at 4 16 30 PM" src="https://user-images.githubusercontent.com/2486846/74686907-f4173400-51a0-11ea-8563-25060de563b5.png">
<img width="1267" alt="Screen Shot 2020-02-17 at 4 16 48 PM" src="https://user-images.githubusercontent.com/2486846/74686908-f5486100-51a0-11ea-9915-ff85ca23cc9a.png">

**Mobile**
<img width="733" alt="Screen Shot 2020-02-17 at 4 17 50 PM" src="https://user-images.githubusercontent.com/2486846/74686948-190ba700-51a1-11ea-9fa1-82068754cd83.png">
<img width="734" alt="Screen Shot 2020-02-17 at 4 17 58 PM" src="https://user-images.githubusercontent.com/2486846/74686950-190ba700-51a1-11ea-96c1-d86c05e15ff5.png">


~Note: The masthead is not going to look very good until #491 gets merged and everything is scaled correctly.~

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. Run `npm run build:sass` and `npm run build:js`
2. Run `npm test` and confirm there are no new errors.
3. Confirm that the [menu block](http://swshumsci.suhumsci.loc/admin/structure/block/manage/humsci_colorful_main_menu) has the following settings
    <img width="470" alt="Configure_block___SWS_H_S" src="https://user-images.githubusercontent.com/2486846/74687240-d8605d80-51a1-11ea-9023-b9df2bf05e9e.png">
4. Confirm that all components of the masthead behave as they should:
    - [x] The lockup links to the homepage
    - [x] The site search works
    - [x] The menu dropdowns work on both at desktop and mobile sizes
    - [x] The mobile navigation works

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
